bangazonapi/views/paymenttype.py

```diff
@@ -81,10 +81,10 @@ def list(self, request):
         """Handle GET requests to payment type resource"""
         payment_types = Payment.objects.all()
 
-        customer_id = self.request.query_params.get('customer', None)
+        customer = request.auth.user
 
-        if customer_id is not None:
-            payment_types = payment_types.filter(customer__id=customer_id)
+        if customer is not None:
+            payment_types = payment_types.filter(customer__id=customer.id)
```

Comment:
Your filter is not working properly because your 'customer' variable isn't actually holding a customer. It's holding a user.

Reply:
Thanks for catching that, the user and customer id's being off by one have gotten me confused. 
I grabbed the customer object that corresponded with the current user, matching on user_id, and then pulled the customer.id to filter the payment types.